docky/common/project.py

```diff
@@ -66,8 +66,14 @@ def _get_services_info(self):
                     if self.project.name in x.attrs["Name"])
         infos = {}
         for serv in services:
-            local = "%s_local" % self.project.name
-            ip = serv.attrs["NetworkSettings"]["Networks"][local].get("IPAddress", "")
+            proj_key = [
+                x for x in serv.attrs["NetworkSettings"]["Networks"].keys()
+                if self.project.name in x]
+            proj_key = proj_key and proj_key[0] or False
+            if not serv.attrs["NetworkSettings"]["Networks"].get(proj_key):
+                continue
+            ip = serv.attrs["NetworkSettings"]["Networks"][proj_key].get(
+                "IPAddress", "")
             info = {
```

Comment:
you can use a try catch or oneliner: 

ip = serv.attrs["NetworkSettings"]["Networks"].get(proj_key, {}).get("IPAddress", "")

Reply:
you 're right

Reply:
I didn't read carefully but if proj_key doesn't exists then `continue` above allow to skip your mentionned code. Then no need try catch, don't you think ?

Reply:
3 way of doing it:
```py
try:
  ip = serv.attrs["NetworkSettings"]["Networks"][proj_key]["IPAddress]
except KeyError :
  continue
```

```py
ip = serv.attrs["NetworkSettings"]["Networks"].get(proj_key, {}).get("IPAddress", "")
```


```py
 if not serv.attrs["NetworkSettings"]["Networks"].get(proj_key):
                continue
 ip = serv.attrs["NetworkSettings"]["Networks"][proj_key].get("IPAddress", "")
```

1 or 2 seems more pythonic to me.

Reply:
IMHO 1/ is hard to read, then not so pythonic
2/ will fail because if proj_key is False then it'll respond: `boolean object have no attribute get`
3/ is my proposal, easy to read and understand

Reply:
In fact I prefer `continue` because it's better to only display services when full info are available